tools/benchmark/src/Configuration.ts

```diff
@@ -209,20 +209,57 @@ export interface BenchmarkTimingOptions {
 }
 
 /**
- * Set of options that can be provided to a benchmark. These options generally align with the BenchmarkJS options type;
- * you can see more documentation {@link https://benchmarkjs.com/docs#options | here}.
+ * Synchronous operations that can be performed on a per-batch basis.
+ *
+ * @remarks
+ * If you need to perform asynchronous operations, use {@link BenchmarkAsyncFunction} and {@link OnBatchAsync}.
+ *
  * @public
  */
 export interface OnBatch {
 	/**
-	 * Executes before the start of each batch. This has the same semantics as benchmarkjs's `onCycle`:
+	 * Executes synchronously before the start of each batch.
+	 * This has the same semantics as benchmarkjs's `onCycle`:
 	 * https://benchmarkjs.com/docs/#options_onCycle
 	 *
 	 * @remarks
 	 * Beware that batches run `benchmarkFn` more than once: a typical micro-benchmark might involve 10k
 	 * iterations per batch.
 	 */
 	beforeEachBatch?: () => void;
+
+	/**
+	 * Use {@link BenchmarkAsyncFunction} and {@link OnBatchAsync} if you need an async hook.
+	 */
+	beforeEachBatchAsync?: never;
+}
+
+/**
+ * Operations (synchronous or asynchronous) that can be performed on a per-batch basis.
+ *
+ * @remarks
+ * Not compatible with synchronous {@link BenchmarkSyncFunction}.
+ *
+ * @public
+ */
+export interface OnBatchAsync {
```

Comment:
`OnBatch` and `OnBatchAsync` leave me a bit confused. Is it strictly necessary to have `beforeEachBatchAsync?: never;` in the former? And should `beforeEachBatch?: () => void;` really be deprecated on the latter? Seems like we allow both to be defined (since we call both in the "synthetic" `beforeAfterBatch` we craft inside `validateBenchmarkArguments`.

Reply:
The key feature is to block synchronous benchmark with asynchronous beforeEachBatch. I tried a number of strategies, this was the best I came up with.

If you can see a simpler way to keep the tests passing, I'm all for it. Or maybe that requirement I held to isn't worth it? Seems like it to me but open to discussion.

Reply:
I deprecated the one in the Async type because it's unnecessary, as you pointed out. Later someone could migrate usages to the new Async function and remove that one.